### PR TITLE
fix(eks): addManifest can accept `any` but only works if a map is passed

### LINKED ISSUE #11483: [aws-eks] addManifest should accept `Record<string, any>[]` instead of `any[]` since it must actually be a JSON map
### Reproduction Steps

I am trying to apply a simple K8s manifest to create a namespace in the Fargate cluster, but while execution it fails to apply
```csharp
var nsManifest = "{\"apiVersion\": \"v1\", \"kind\": \"Namespace\", \"metadata\": { \"name\": \"examples\"}}";
cluster.AddManifest("ns", nsManifest);
```

### What did you expect to happen?

Provided manifest gets applies to the cluster and a new namespace gets created.

### What actually happened?

Received below error while running the **cdk deploy** step

> Custom::AWSCDK-EKS-KubernetesResource | eks-ew1-examples-dev/manifest-ns/Resource/Default (eksew1examplesdevmanifestns550D618B) Failed to create resource. Error: b'error: error validating "/tmp/manifest.yaml": error validating data: invalid object to validate; if you choose to ignore these errors, turn validation off with --validate=false\n'

On checking the generated template fragment, i see the manifest is created with extra backslashes like below.
```
"Manifest": "[\"{\\\"apiVersion\\\": \\\"v1\\\", \\\"kind\\\": \\\"Namespace\\\", \\\"metadata\\\": { \\\"name\\\": \\\"examples\\\"}}\"]",
```

### Environment

  - **CDK CLI Version :** 1.73.0
  - **Framework Version:**
  - **Node.js Version:** v14.15.0
  - **OS:** Windows 10
  - **Language (Version):** .NETCore3.1

### Other

I also tried to create custom structs and tried to apply it instead of passing the manifest json as string like below
```csharp
 cluster.AddManifest("ns", new K8sManifest
{
    ApiVersion = "v1",
    Kind = "Namespace",
    Metadata = new Metadata
    {
        Name = "examples"
     }
});

public struct K8sManifest
{
   public string ApiVersion { get; set; }
   public string Kind { get; set; }
   public Metadata Metadata { get; set; }
}

public struct Metadata
{
    public string Name { get; set; }
}
```
But the **cdk synth** process fails with below error

> Unhandled exception. System.ArgumentException: Could not infer JSII type for .NET type 'K8sManifest' (Parameter 'type')
   at Amazon.JSII.Runtime.Services.Converters.FrameworkToJsiiConverter.InferType(IReferenceMap referenceMap, Type type)

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by ayush987goyal**
Change to `Record` to prevent users from accidentally passing a string (or anything else for that matter).

Closes #11483

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by iliapolo** [comment]
@ayush987goyal Thanks for this, needs some tests though.

**Comment by ayush987goyal** [comment]
I was wondering how would I add a test to verify an incorrect manifest type being passed (because it would not get compiled by tsc). Any pointers on this?

**Comment by iliapolo** [comment]
@ayush987goyal - ha, good point :) I've adde the test exemption label.

Let me double check this doesn't have any delicate edges we missed.

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 1]
```diff
@@ -641,7 +641,7 @@ abstract class ClusterBase extends Resource implements ICluster {
    * @param manifest a list of Kubernetes resource specifications
    * @returns a `KubernetesResource` object.
    */
-  public addManifest(id: string, ...manifest: any[]): KubernetesManifest {
+  public addManifest(id: string, ...manifest: Record<string, any>[]): KubernetesManifest {
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:644]: The type should also be changed here:

https://github.com/aws/aws-cdk/blob/2e067d7d00b10b4c5e26665386c0a86340a83379/packages/%40aws-cdk/aws-eks/lib/k8s-manifest.ts#L42

**Review by iliapolo — CHANGES_REQUESTED** [review]

**Comment by iliapolo** [comment]
@eladb Can you think of any reason not to do this change?

**Comment by eladb** [comment]
@iliapolo I'd just make sure the .jsii output looks ok

**Comment by iliapolo** [comment]
@eladb looks good.

```json
          "name": "addManifest",
          "parameters": [
            {
              "docs": {
                "summary": "logical id of this manifest."
              },
              "name": "id",
              "type": {
                "primitive": "string"
              }
            },
            {
              "docs": {
                "summary": "a list of Kubernetes resource specifications."
              },
              "name": "manifest",
              "type": {
                "collection": {
                  "elementtype": {
                    "primitive": "any"
                  },
                  "kind": "map"
                }
              },
              "variadic": true
            }
          ],
```

**Review by iliapolo — APPROVED** [review]

**Pull request merged**